pre_commit_hooks/check_case_conflict.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import itertools
+import sys
+
+from plumbum import local
+
+
+def find_conflicting_filenames(filenames):
+    # Find all files already in the repo
+    repo_filenames = set(local['git'](
+        'ls-files', '--exclude-standard').splitlines())
+
+    # Compare the files pre-commit says are modified against all files in the
+    # repo
+    pairs = itertools.product(filenames, repo_filenames)
+
+    retv = 0
+    for pair in pairs:
+        if pair[0] != pair[1] and pair[0].lower() == pair[1].lower():
+            print('{0} and {1} differ only in case.'.format(pair[0], pair[1]))
+            retv = 1
+
+    return retv
+
+
+def main(argv=None):
+    argv = argv if argv is not None else sys.argv[1:]
```

Comment:
so I've found that parser.parse_args(None) will magically use sys.argv, so you can actually drop this line